libsolidity/codegen/MultiUseYulFunctionCollector.h

```diff
@@ -40,14 +41,28 @@ class MultiUseYulFunctionCollector
 	/// cases.
 	std::string createFunction(std::string const& _name, std::function<std::string()> const& _creator);
 
-	/// @returns concatenation of all generated functions.
+	/// Helper function that uses @a _creator to create a function and add it to
+	/// @a m_requestedFunctions if it has not been created yet and returns @a _name in both
+	/// cases.
+	std::string createExternallyUsedFunction(std::string const& _name, std::function<std::string()> const& _creator);
+
+	/// Manually mark a function as externally used.
+	void markAsExternallyUsed(std::string const& _name)
+	{
+		m_externallyUsedFunctions.insert(_name);
+	}
+
+	/// @returns concatenation of all generated functions and a set of the
+	/// externally used functions.
 	/// Clears the internal list, i.e. calling it again will result in an
 	/// empty return value.
-	std::string requestedFunctions();
+	std::pair<std::string, std::set<std::string>> requestedFunctions();
```

Comment:
Since it's "moving out" some previously generated data, wouldn't it make sense to have a signature like this:

```cpp
std::pair<std::string, std::set<std::string>>&& requestedFunctions();
```

Reply:
It's returning temporaries by value, I think it's fine or might actually even need to be like this - but it'll look different in a second anyways.